third-party/CMakeLists.txt

```diff
@@ -19,6 +19,12 @@
 
 message("Entering ${CMAKE_CURRENT_SOURCE_DIR}")
 
-add_subdirectory(CLI11 EXCLUDE_FROM_ALL)
-add_subdirectory(fmt EXCLUDE_FROM_ALL)
-include(./xxHash.cmake)
+if(NOT USE_SYSTEM_CLI11)
+  add_subdirectory(CLI11 EXCLUDE_FROM_ALL)
+endif()
+if(NOT USE_SYSTEM_FMT)
+  add_subdirectory(fmt EXCLUDE_FROM_ALL)
+endif()
+if(NOT USE_SYSTEM_XXHASH)
+  include(./xxHash.cmake)
+endif()
```

Comment:
why did you scatter your changes into several files.
The options from the topmost CMakeLists.txt could have been in here and so could have been the additions to `core/cmake/BareosFindAllLibraries.cmake`.

Please fix the issues I mentioned above and consolidate the changes into one place.